language/stdlib/modules/vector.mvir

```diff
@@ -106,6 +116,11 @@ module Vector {
     return false;
   }
 
+  // Return true if unrestricted value `e` is in the vector `v`
+  public contains_unrestricted<Element: unrestricted>(v: &Self.T<Element>, e: Element): bool {
```

Comment:
Why have this API if the user can just borrow the value?

Reply:
Keep in mind, in the source language, you can borrow non-locals, as it will create a temporary local for you

Reply:
You can't borrow a value on the stack. It would be nice to able to write `v.contains(3)` without needing a local (even if the compiler is happy to generate one for you).

Reply:
You still need the local though, the implementation of `contains_unrestricted` is just going to borrow the value. So the local is moved from your function to `contains_unrestricted`. And the overall gas cost remains the same 

In short, 
`v.contains(&3)` is going to cost the same as `v.contains_unrestricted(3)`

In many cases, `contains_unrestricted` is going to cost more due to a copy

Reply:
> And the overall gas cost remains the same

The gas cost is the same, but the code size is smaller if the local lives in the callee instead of in each caller.

I see your point, though; to get the advantages I was pitching you'd really want to reimplement the body of `contains` instead of just calling it.

Reply:
I don't think you'll get any advantages though. You'll have to copy O(n) times